src/core/timer.c

```diff
@@ -332,7 +364,10 @@ static void add_random(Timer *t, usec_t *v) {
         if (*v == USEC_INFINITY)
                 return;
 
-        add = random_u64() % t->random_usec;
+        if (t->fixed_random_delay)
+                add = timer_get_fixed_delay(t);
+        else
+                add = random_u64() % t->random_usec;
```

Comment:
nitpick: i'd probably move the `% t->random_usec` thing out of timer_get_fixed_delay(), and apply that here for both cases. i.e.

```c
add = (t->fixed_random_delay ? 
                timer_get_fixed_hash(t) : 
                random_u64()) % t->random_usec;
```

i.e. just to unify the two codepaths a tiny bit.

but doesn't really matter